### ISSUE TITLE
Waiting for service worker

### ISSUE THREAD

**Issue by erezsh** [NekitCorp/chrome-extension-svelte-typescript-boilerplate#12]
Interacting with the boilerplate extension after letting the service worker to become inactive, we are greeting with the aforementioned message, which doesn't go away until we close and open the page again -

![image](https://user-images.githubusercontent.com/731464/215346593-2dedcb0a-1157-46d6-a94a-a9bfd349c117.png)

![image](https://user-images.githubusercontent.com/731464/215346620-c8776fd0-5712-4053-937d-aa3d94597858.png)

I think the desired behavior would be to hide this message once the worker is done loading.

**Comment by NekitCorp** [comment]
It's issue from `crxjs`: https://github.com/crxjs/chrome-extension-tools/issues/449

Follow the link to find out why this happens and what it affects. In short, this error only occurs during development and due to the fact that you turn off the development dev `Vite` server.

**Comment by erezsh** [comment]
This happens while the server is running.

The worker starts immediately after displaying that message, and reloading the page gets rid of the message. But I think the reload should be automatic?

**Comment by NekitCorp** [comment]
Yes, when you first start the server dev, there should not be such an error. Do you have a sample project that can reproduce this issue? Or is it reproduced even if you just clone this boilerplate?

**Comment by NekitCorp** [comment]
I don't quite understand what the problem is, could you send me the steps to reproduce and the expected behavior please? And also please tell me this problem is reproduced in a clean fork `chrome-extension-svelte-typescript-boilerplate`?

**Comment by NekitCorp** [comment]
And once again I repeat that this error occurs **only during development**. In production, this error will not occur. It seems to me that the case with disabling the service worker is quite rare (because, for example, chrome works in the backround mode even when the browser is closed and the service worker continues to work), so simply reopening the extension to solve this problem in development mode is enough. In general, after reopening, the extension will already know about restarting the server and worker.

![Screen Recording 2023-01-30 at 15 40 13](https://user-images.githubusercontent.com/15874982/215479726-dd4e824f-ada9-4689-bda1-ec9d9d820916.gif)

**Comment by erezsh** [comment]
Yes, this happens with a clean fork.

Steps are simple:

1. Run extension
2. Wait for service worker to become idle (takes about a minute)
3. Open options / popup, and get this message
4. Opening options / popup again immediately after will work as expected.

I understand this only happens in dev mode.

Whether it's enough or not is, I suppose, subjective. But I think that if the boilerplate knows to show this error page, it shouldn't be that hard to just wait for the worker to be ready and reload? It would definitely be a nicer dev experience.

**Comment by NekitCorp** [comment]
Oh, thanks, i get what you mean. But unfortunately, I can’t influence this logic in this boilerplate, because. this is the package logic https://github.com/crxjs/chrome-extension-tools

I think you can bring this idea to this issue: https://github.com/crxjs/chrome-extension-tools/issues/449
